win32/win_fopen.c

```diff
@@ -0,0 +1,148 @@
+//
+//  Copyright(C) 2021 Roman Fomin
+//
+//  This program is free software; you can redistribute it and/or
+//  modify it under the terms of the GNU General Public License
+//  as published by the Free Software Foundation; either version 2
+//  of the License, or (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+// DESCRIPTION:
+//      unicode paths for fopen() on Windows
+
+#include "win_fopen.h"
+
+#ifdef _WIN32
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#include <stdlib.h>
+#include <direct.h>
+
+static wchar_t* ConvertToUtf8(const char *str)
+{
+    wchar_t *wstr = NULL;
+    int wlen = 0;
+
+    wlen = MultiByteToWideChar(CP_UTF8, 0, str, -1, NULL, 0);
+
+    if (!wlen)
```

Comment:
This should also print an error message, either here or in the callers. That way the user is not left wondering what happened when something goes wrong.

Reply:
I think a reasonable error message here is "path not found" and should be printed on the side of the `fopen()` caller.

Reply:
I don't think so. If the user sees "path not found" because there was a conversion error and they know that the path they specified does exist they're going to be very confused. A conversion error should explicitly say that something went wrong in the conversion, not some upstream call.

Reply:
I agree with turol - this should be a relatively uncommon situation and printing an error message to help debug seems like it could be useful.